system_test/e2e/joining.py

```diff
@@ -158,6 +160,48 @@ def test_raft_adv(self):
     l2 = self.n2.wait_for_leader()
     self.assertEqual(l0, l2)
 
+class TestSingleNodeJoin(unittest.TestCase):
+  def setUp(self):
+    # Setup single node
+    self.n0 = Node(RQLITED_PATH, '0')
+    self.n0.start()
+    self.n0.wait_for_leader()
+    
+    # Create a potential join target node
+    self.n1 = Node(RQLITED_PATH, '1')
+    self.n1.start()
+    self.n1.wait_for_leader()
+
+  def test_single_node_join_fails(self):
+    '''Test that a single-node cluster leader cannot join another cluster'''
+    # Check that n0 is a single-node cluster
+    nodes = self.n0.nodes()
+    self.assertEqual(len(nodes), 1)
+    
+    # Attempting to join n0 to n1 should fail
+    n0_restart = Node(RQLITED_PATH, '0', dir=self.n0.dir)
```

Comment:
This could be problematic because the first node is running and we want the node to restart with exactly the same attributes (except with a new join flag):

Let's do this instead:
- call self.n0.stop(graceful=True) to stop the node.
- call self.n0.start() - but set the join parameter. However this start operation should throw an exception, since the process should exit. So the test fails if the restart doesn't throw exception.

Reply:
Perhaps we should add a helper method to the Python Node class, which returns true if the process is running, false others.

def running(self):
    # Return true if self.process is actually an existing process in the process table.